interface/billing/sl_eob_search.php

```diff
@@ -81,38 +86,70 @@ function upload_file_to_client($file_to_send) {
   sleep(1);
 }
 function upload_file_to_client_pdf($file_to_send) {
-//Function reads a text file and converts to pdf.
+  //Function reads a HTML file and converts to pdf.
 
   global $STMT_TEMP_FILE_PDF;
-  $pdf = new Cezpdf('LETTER');//pdf creation starts
-  $pdf->ezSetMargins(36,0,36,0);
-  $pdf->selectFont('Courier');
-  $pdf->ezSetY($pdf->ez['pageHeight'] - $pdf->ez['topMargin']);
-  $countline=1;
-  $file = fopen($file_to_send, "r");//this file contains the text to be converted to pdf.
-  while(!feof($file))
-   {
-    $OneLine=fgets($file);//one line is read
-	 if(stristr($OneLine, "\014") == true && !feof($file))//form feed means we should start a new page.
-	  {
-	    $pdf->ezNewPage();
-	    $pdf->ezSetY($pdf->ez['pageHeight'] - $pdf->ez['topMargin']);
-		str_replace("\014", "", $OneLine);
-	  }
-	
-	if(stristr($OneLine, 'REMIT TO') == true || stristr($OneLine, 'Visit Date') == true || stristr($OneLine, 'Future Appointments') == true || stristr($OneLine, 'Current') == true)//lines are made bold when 'REMIT TO' or 'Visit Date' is there.
-	 $pdf->ezText('<b>'.$OneLine.'</b>', 12, array('justification' => 'left', 'leading' => 6));
-	else
-	 $pdf->ezText($OneLine, 12, array('justification' => 'left', 'leading' => 6));
-	 
-	$countline++;
-   }
-	
-	$fh = @fopen($STMT_TEMP_FILE_PDF, 'w');//stored to a pdf file
+  global $srcdir;
+
+  if ($GLOBALS['statement_appearance'] == '1') {
+    require_once("$srcdir/html2pdf/vendor/autoload.php");
+    $pdf2 = new HTML2PDF ($GLOBALS['pdf_layout'],
+    $GLOBALS['pdf_size'],
+    $GLOBALS['pdf_language'],
+                           true, // default unicode setting is true
+                           'UTF-8', // default encoding setting is UTF-8
+                           array($GLOBALS['pdf_left_margin'],$GLOBALS['pdf_top_margin'],$GLOBALS['pdf_right_margin'],$GLOBALS['pdf_bottom_margin']),
+                           $_SESSION['language_direction'] == 'rtl' ? true : false
+                           );
+    ob_start();
+    echo readfile($file_to_send, "r");//this file contains the HTML to be converted to pdf.
+    //echo $file;
+    $content = ob_get_clean();
+
+    // Fix a nasty html2pdf bug - it ignores document root!
+    global $web_root, $webserver_root;
+    $i = 0;
+    $wrlen = strlen($web_root);
+    $wsrlen = strlen($webserver_root);
+    while (true) {
+      $i = stripos($content, " src='/", $i + 1);
+      if ($i === false) break;
+      if (substr($content, $i+6, $wrlen) === $web_root &&
+        substr($content, $i+6, $wsrlen) !== $webserver_root) {
+        $content = substr($content, 0, $i + 6) . $webserver_root . substr($content, $i + 6 + $wrlen);
+      }
+    }
```

Comment:
This code looks a bit scary :)
(not saying it shouldn't be there, though)
Just out of curiosity, is this an issue everywhere that html2pdf is used(such as in the patient pdf report)?